graph/config_io.py

```
@@ -0,0 +1,321 @@
+
+def validate_config_dict(updates: dict[str, Any]) -> tuple[bool, str]:
+    """Validate without persisting. Returns (ok, error-message).
+
+    Catches type mismatches and obvious range errors before we touch
+    disk or rebuild the graph.
+    """
+    try:
+        model = updates.get("model", {})
+        temp = float(model.get("temperature", 0.2))
+        if not 0.0 <= temp <= 2.0:
+            return False, f"temperature must be 0.0–2.0, got {temp}"
```

Ruff (0.15.10)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)